frontend/app/services/MembersDataAPI.scala

```diff
@@ -46,19 +47,32 @@ object MembersDataAPI {
     (JsPath \ "details").read[String]
   )(ApiError)
 
-  case class Helper(cookies: Seq[Option[Cookie]]) extends WebServiceHelper[Attributes, ApiError] {
-    override val wsUrl: String = Config.membersDataAPIUrl
-    override def wsPreExecute(req: Request.Builder): Request.Builder = req.addHeader(
-      "Cookie", cookies.flatten.map(c => s"${c.name}=${c.value}").mkString("; "))
+  implicit val behaviourReads: Reads[Behaviour] = (
+    (JsPath \ "userId").read[String] and
+    (JsPath \ "activity").read[String] and
+    (JsPath \ "lastObserved").read[String]
+  )(Behaviour.apply _)
 
+  case class AttributeHelper(cookies: Seq[Option[Cookie]]) extends WebServiceHelper[Attributes, ApiError] {
+    override val wsUrl: String = Config.membersDataAPIUrl
+    override def wsPreExecute(req: Request.Builder): Request.Builder = {
+      req.addHeader("Cookie", cookies.flatten.map(c => s"${c.name}=${c.value}").mkString("; "))
+    }
     override val httpClient: LoggingHttpClient[Future] = RequestRunners.loggingRunner(MembersDataAPIMetrics)
+  }
 
+  case class BehaviourHelper(cookies: Seq[Option[Cookie]]) extends WebServiceHelper[Behaviour, ApiError] {
+    override val wsUrl: String = Config.membersDataAPIUrl
+    override def wsPreExecute(req: Request.Builder): Request.Builder = {
+      req.addHeader("Cookie", cookies.flatten.map(c => s"${c.name}=${c.value}").mkString("; "))
+    }
+    override val httpClient: LoggingHttpClient[Future] = RequestRunners.loggingRunner(MembersDataAPIMetrics)
   }
```

Comment:
Try to avoid repetition like this! `AttributeHelper` and `BehaviourHelper` are identical apart from the type parameter of the `WebServiceHelper` they extend, so it should be possible to make one class with that implementation gunk that takes that differing type as it's own type parameter, and have two one-line instances of it that differ only in that parameter.